op-e2e/actions/user_test.go

```diff
@@ -30,42 +51,75 @@ type hardforkScheduledTest struct {
 // - wait 1 week + 1 second
 // - finalize withdrawal on L1
 func TestCrossLayerUser(t *testing.T) {
-	zeroTime := hexutil.Uint64(0)
 	futureTime := hexutil.Uint64(20)
 	farFutureTime := hexutil.Uint64(2000)
-	tests := []hardforkScheduledTest{
-		{name: "NoRegolith", regolithTime: nil, activateRegolith: false, deltaTime: nil, activateDelta: false},
-		{name: "NotYetRegolith", regolithTime: &farFutureTime, activateRegolith: false, deltaTime: nil, activateDelta: false},
-		{name: "RegolithAtGenesis", regolithTime: &zeroTime, activateRegolith: true, deltaTime: nil, activateDelta: false},
-		{name: "RegolithAfterGenesis", regolithTime: &futureTime, activateRegolith: true, deltaTime: nil, activateDelta: false},
-		{name: "NoDelta", regolithTime: &zeroTime, activateRegolith: true, deltaTime: nil, activateDelta: false},
-		{name: "NotYetDelta", regolithTime: &zeroTime, activateRegolith: true,
-			deltaTime: &farFutureTime, activateDelta: false},
-		{name: "DeltaAtGenesis", regolithTime: &zeroTime, activateRegolith: true,
-			deltaTime: &zeroTime, activateDelta: true},
-		{name: "DeltaAfterGenesis", regolithTime: &zeroTime, activateRegolith: true,
-			deltaTime: &futureTime, activateDelta: true},
+
+	forks := []string{
+		"regolith",
+		"canyon",
+		"delta",
+		"ecotone",
+		"fjord",
 	}
-	for _, test := range tests {
-		test := test // Use a fixed reference as the tests run in parallel
-		t.Run(test.name, func(gt *testing.T) {
-			runCrossLayerUserTest(gt, test)
+	for i, fork := range forks {
+		i := i
+		fork := fork
+		t.Run("fork_"+fork, func(t *testing.T) {
+			t.Run("at_genesis", func(t *testing.T) {
+				tc := hardforkScheduledTest{}
+				for _, f := range forks[:i+1] { // activate, all up to and incl this fork, at genesis
+					*tc.fork(f) = new(hexutil.Uint64)
```

Comment:
That hurts my head a bit. Maybe it's worth `fork` accepting the block the fork should activate at and actually setting the variable instead of just returning a reference to it?

or possibly just have a `scheduleFork` method that does that if it's useful to be able to get a reference to the fork variable.

Reply:
changed it to use getters/setters